### ISSUE TITLE
Serialization strategy?

### ISSUE THREAD

**Issue by piedoom** [zkat/big-brain#104]
I'm wondering if it's possible/within the scope of this crate to allow serialization (specifically of the data in a `ThinkerBuilder`). I can see it being useful to persist AI to a save, or prototype quickly by adjusting values. I can think of a few alternatives (mainly writing my own intermediate struct to serialize/deserialize the `ThinkerBuilder` to) but this seems like the neatest option

**Comment by zkat** [comment]
While `Thinker`s &co are fundamentally stateless, the really good reason for having serializable builders is so you can build up your AI logic using external tools/node-based graphical builders, or even bevy's upcoming scene editor.

I think things are in too much flux to make any real guarantees, but I think it makes sense for this library to have explicit compatibility with bevy's scene definition format once it's ready.

**Comment by piedoom** [comment]
Makes sense, thanks for the response! I'll rely on some bespoke solutions til Bevy's new scene stuff gets finalized.
